### PR TITLE
Pass bigtable::TableAdmin by value in sample snippets.

### LINKED ISSUE #645: Change table admin examples to pass bigtable::TableAdmin object by value
These examples were created when the TableAdmin objects were non-copyable, they do not reflect best-practices now that the objects can be freely copied.

### PR DESCRIPTION
**Pull request by coryan**
This is a simpler model, no need to complicate the code for users.
This fixes #645.

### CODE REVIEW

File: google/cloud/bigtable/examples/bigtable_samples.cc [hunk 1]
```diff
@@ -30,7 +30,7 @@ namespace {
 const std::string MAGIC_ROW_KEY = "key-000009";
 
 //! [create table]
-void CreateTable(google::cloud::bigtable::TableAdmin& admin,
+void CreateTable(google::cloud::bigtable::TableAdmin admin,
```
Comment (mbrukman) [inline google/cloud/bigtable/examples/bigtable_samples.cc]: What's the benefit of passing it by value, when it has so many [member fields](https://github.com/GoogleCloudPlatform/google-cloud-cpp/blob/8d2288a7f9d68c9eb66c01abc7fa5e5691d83f19/google/cloud/bigtable/internal/table_admin.h#L223-L229)?

I realize performance is not an issue for a simple example, but we're passing the `table_id` (a `std::string`) by const reference, so as a code reader, I'm confused why the admin class wouldn't similarly be passed by reference.

Is the suggestion that sample code having fewer references is easier to read? Or is the suggestion that no one should ever pass around a `TableAdmin` object by reference? Are there cases where users should prefer to pass it by reference?

I imagine we're going to get feedback and/or PRs to add references to these calls if we were to remove them.

On the one hand, non-const references are against the [Google C++ style guide](https://google.github.io/styleguide/cppguide.html#Reference_Arguments).

Comment (coryan) [reply google/cloud/bigtable/examples/bigtable_samples.cc]: We want to highlight that these are canonical C++ objects, and application developers should feel free to use copy semantics, or move semantics, or references.  The decision is up to them, depending on the performance considerations of their application.

Showing reference-only examples sends (IMHO) the signal that these are not [CopyAssignable](http://en.cppreference.com/w/cpp/concept/CopyAssignable).  Maybe I am reading too much into it though.

Comment (mbrukman) [reply google/cloud/bigtable/examples/bigtable_samples.cc]: OK, then I recommend @deepankarsharma and @garye take a look; I'll bow out.

**Comment by coryan** [comment]
@deepankarsharma can you look at this one too?  Very similar to #701.

**Comment by coryan** [comment]
Ping.

**Pull request merged**